src/Orchard.Environment.Shell/RunningShellTable.cs

```diff
@@ -33,8 +33,11 @@ public void Add(ShellSettings settings)
                 }
 
                 var hostAndPrefix = GetHostAndPrefix(settings);
+                if (_shellsByHostAndPrefix.ContainsKey(hostAndPrefix))
+                    return false;
                 _shellsByHostAndPrefix.Add(hostAndPrefix, settings);
```

Comment:
I agree. 
Now Add and Update methods are almost the same. Won't be cleaner if we remove Add method and only offer an Update method that does just what Add is doing now?